### PR TITLE
Add root cause to our logging

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@calvinjia This change gives us more information to work with in case of a deeply nested exception. PTAL

File: core/server/common/src/main/java/alluxio/ProcessUtils.java [hunk 1]
```diff
@@ -37,11 +37,13 @@ public static void run(Process process) {
       System.exit(0);
     } catch (Throwable t) {
       LOG.error("Uncaught exception while running {}, stopping it and exiting.", process, t);
+      LOG.error("Root cause of the exception: ", ExceptionUtils.getRootCause(t));
```
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/ProcessUtils.java]: Can we include this in the previous log message?

Comment (yuzhu) [reply core/server/common/src/main/java/alluxio/ProcessUtils.java]: done

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Comment by apc999** [comment]
@calvinjia @yuzhu this pr introduced a side-effect that the stack trace is lost from the log. Note that the original logging will output the entire exception including stack trace of `t`:
```
LOG.error("Uncaught exception while running {}, stopping it and exiting.", process, t)
```
After this PR, one will see sth like, to me it actually becomes harder to debug.
```
‘2019-07-03 14:58:01,065 ERROR ProcessUtils - Uncaught exception while stopping Alluxio worker, simply exiting. Exception java.lang.NullPointerException, Root Cause null‘ 
```
I would put back `t` in the end as the last arg

**Pull request merged**